examples/pytorch/rgcn-hetero/README.md

```diff
@@ -0,0 +1,54 @@
+# Relational-GCN
+
+* Paper: [https://arxiv.org/abs/1703.06103](https://arxiv.org/abs/1703.06103)
+* Author's code for entity classification: [https://github.com/tkipf/relational-gcn](https://github.com/tkipf/relational-gcn)
+* Author's code for link prediction: [https://github.com/MichSchli/RelationPrediction](https://github.com/MichSchli/RelationPrediction)
+
+The preprocessing is slightly different from the author's code. We directly load and preprocess
+raw RDF data. For AIFB, BGS and AM,
+all literal nodes are pruned from the graph. For AIFB, some training/testing nodes
+thus become orphan and are excluded from the training/testing set. The resulting graph
+has fewer entites and relations. As a reference (numbers include reverse edges and relations):
+
+| Dataset | #Nodes | #Edges | #Relations | #Labeled |
+| --- | --- | --- | --- | --- |
+| AIFB | 8,285 | 58,086 | 90 | 176 |
+| AIFB-hetero | 1711 | 31,400 | 32 | 135 |
+| MUTAG | 23,644 | 148,454 | 46 | 340 |
+| MUTAG-hetero | 27,163 | 148,100 | 46 | 340 |
+| BGS | 333,845 | 1,832,398 | 206 | 146 |
+| BGS-hetero | 94,806 | 672,884 | 96 | 146 |
+| AM | 1,666,764 | 11,976,642 | 266 | 1000 |
+| AM-hetero | 881,680 | 5,668,682 | 96 | 1000 |
+
+### Dependencies
+* PyTorch 1.0+
+* requests
+* rdflib
+
+```
+pip install requests torch rdflib pandas
+```
+
+Example code was tested with rdflib 4.2.2 and pandas 0.23.4
+
+### Entity Classification
+AIFB: accuracy 100.0% (DGL), 95.83% (paper)
```

Comment:
100%?  Very suspicious...  Does the dataset have leakage?

Reply:
So I checked that there seems to be no data leakage during testing, but the training + validation + test set is tiny (151 in total).

Reply:
I removed all the literal nodes during preprocessing, making some training and testing nodes dangling nodes. I checked the dataset. They look like this:
```
Person:
  Phone: ""
  Address: ""
```
"Phone" and "Address" are all literal nodes and are removed. For this dataset, I could add them back. The other dataset does not have this issue.

Reply:
For the question about whether this is the same scale as the original one. The answer is NO due to the first embedding layer. The reason is that we cannot use the basis decomposition for the first embedding layer. The original input is an one-hot encoding of size N (number of nodes) for each node, which does not make sense when the number of nodes of different types varies.

Reply:
If the scale is smaller from the original one by more than an order of magnitude, then I doubt if the experiment is meaningful...  We can still keep the code, but probably I'll just remove the number on AMiner.

Reply:
The model scale is not smaller, but larger since there is no basis decomposition for the first embedding layer. Also, where is AMiner number? The AM dataset is for Amsterdam Museum.

Reply:
Sorry I meant AIFB